Tests/Tests/TDMiscTests.m

```diff
@@ -75,5 +75,164 @@ - (void)testEscapeID
+    STAssertEqualObjects(cleanString, @"myscheme://*****@myhost.com:1234/db?all_docs=true",
+                         @"not cleaned: %@", cleanString);
+    
+    testUrl = [NSURL URLWithString:@"http://adam:adamspassword@myhost.com"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"http://*****@myhost.com", @"not clean: %@", cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam:adamspasswordmyhost@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@/db", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@/db", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam@"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam@myhost.com/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@myhost.com/db", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://nothing.to.clean/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://nothing.to.clean/db", @"should not have changed: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://@/db", @"should not have changed: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://:@"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://:@", @"should not have changed: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://:@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://:@/db", @"should not have changed: %@",
+                         cleanString);
+    
+    
+    
+    NSString *errorMessage = @"NSURL did not have both a scheme and host. Suppressing string value for security.";
+    
+    //
+    //these should fail due to typo in the '://', which prevents NSURL from parsing the string.
+    //
+    testUrl = [NSURL URLWithString:@"https:/adam:adamspassword@myhost.com/db"];
+    STAssertEqualObjects(testUrl.scheme, @"https", @"scheme not https?  %@", testUrl.scheme);
+    STAssertNil(testUrl.host, @"host not nil?  %@", testUrl.host);
+    STAssertNil(testUrl.user, @"user not nil?  %@", testUrl.user);
+    STAssertNil(testUrl.password, @"password not nil?  %@", testUrl.password);
```

Comment:
Do you need all the checks for hard-coded values? Testing your own string seems odd.